app/forms/shop_registration_form.rb

```diff
@@ -0,0 +1,21 @@
+class ShopRegistrationForm < AddressForm
+  attr_accessor :name, :description, :phone, :user, :shop, :shop_address
+
+  validates :name, presence: true 
+  validates :description, presence: true
+  validates :phone, presence: true
+
+  def save 
+    return false if invalid?
+    @shop = Shop.create(name: name, description: description, phone: phone, user: user)
+    @shop_address = create_address
```

Comment:
- Don't use instance variable here if you don't re-use it in another function.
- You have check the if params is valid but not user. What if user is invalid and `Shop.create` is failing?

Reply:
Updated:
- Instance variables are changed to local variables
- #save now applies Transactions to create records